### PR TITLE
Improve `param_importances`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

The current `param_importances` add different colors depending on the distribution class that users may not access directly. In addition, single color among distributions is enough to visualize important hyperparameters when we think about the role of this plot -- the distribution does not matter.

See also https://github.com/optuna/optuna/pull/1991#issuecomment-816726700.

## Description of the changes

- use a single color map for both Plotly and Matplolib backends.
- Remove legend from Matplolib backend.

---

```python
import optuna


def objective(trial):
    x = trial.suggest_int("x", 0, 2)
    y = trial.suggest_float("y", -1.0, 1.0)
    z = trial.suggest_float("z", 0.0, 1.5)
    return x ** 2 + y ** 3 - z ** 4


sampler = optuna.samplers.RandomSampler(seed=10)
study = optuna.create_study(sampler=sampler)
study.optimize(objective, n_trials=10)

optuna.visualization.matplotlib.plot_param_importances(study)
```

## Plotly

newplot (1)

## Matplotlib

download

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@himkt Could you review this PR?

**Review by HideakiImamura — APPROVED** [review]
LGTM.

File: optuna/visualization/_param_importances.py [hunk 1]
```diff
@@ -25,16 +19,8 @@
 
     from optuna.visualization._plotly_imports import go
 
-    Blues = plotly.colors.sequential.Blues
+    bar_color = plotly.colors.sequential.Blues[-4]
```
Comment (himkt) [inline optuna/visualization/_param_importances.py:22]: I think making a variable here is no more needed, as we remove `_distribution_colors`.

File: optuna/visualization/matplotlib/_param_importances.py [hunk 2]
```diff
@@ -27,26 +21,10 @@
...
-    }
-    _legend_elements = [
-        Rectangle([0, 0], 0, 0, facecolor=cmap(0), label="Uniform Distribution"),
-        Rectangle([0, 0], 0, 0, facecolor=cmap(0), label="Log Uniform Distribution"),
-        Rectangle([0, 0], 0, 0, facecolor=cmap(0), label="Discrete Uniform Distribution"),
-        Rectangle([0, 0], 0, 0, facecolor=cmap(1), label="Int Uniform Distribution"),
-        Rectangle([0, 0], 0, 0, facecolor=cmap(1), label="Int Log Uniform Distribution"),
-        Rectangle([0, 0], 0, 0, facecolor=cmap(3), label="Categorical Distribution"),
-    ]
+    bar_color = cmap(0)
```
Comment (himkt) [inline optuna/visualization/matplotlib/_param_importances.py:27]: Same as https://github.com/optuna/optuna/pull/2576/files#r613301532.
I think we can pass `cmap(0)` directory to `color` in `ax.barh`.

**Comment by nzw0301** [comment]
@himkt Thank you for the feedbacks! Your concerns have been resolved by cd08914.

**Review by himkt — APPROVED** [review]
LGTM, thank you for the quick fix and improving the experience!

**Pull request merged**